napari/plugins/hub.py

```diff
@@ -97,15 +98,36 @@ def hub_plugin_info(
     return data, is_available_in_conda_forge
 
 
+def _filter_summaries(
+    summaries: List[SummaryDict], plugin_name: str
+) -> Union[SummaryDict, None]:
+    """Returns the plugin summary specified by plugin_name from a list of summaries."""
+
+    found = False
+    cnt = 0
+    while found is False and cnt < len(summaries):
+        if summaries[cnt]['name'].lower().replace('-', '').replace(
+            '_', ''
+        ) == plugin_name.lower().replace('-', '').replace('_', ''):
```

Comment:
Why is this called on every loop round? It should be called only once before loop start